src/continuous_agg.c

```diff
@@ -405,37 +405,64 @@ ts_continuous_agg_find_by_rv(const RangeVar *rv)
 	return ts_continuous_agg_find_by_relid(relid);
 }
 
+static void
+get_rel_by_name(ObjectAddress *addr, const char *schema, const char *name, LOCKMODE mode)
+{
+	Oid relid = InvalidOid;
+	Oid nspid = get_namespace_oid(schema, true);
+	if (OidIsValid(nspid))
+	{
+		relid = get_relname_relid(name, nspid);
+		if (OidIsValid(relid))
+			LockRelationOid(relid, mode);
+	}
+	ObjectAddressSet(*addr, RelationRelationId, relid);
+}
+
+static void
+get_rel_by_hypertable_id(ObjectAddress *addr, int32 hypertable_id, LOCKMODE mode)
+{
+	Oid relid = ts_hypertable_id_to_relid(hypertable_id);
+	if (OidIsValid(relid))
+		LockRelationOid(relid, mode);
+	ObjectAddressSet(*addr, RelationRelationId, relid);
+}
+
 /*
  * Drops continuous aggs and all related objects.
  *
- * These objects are: the user view itself, the catalog entry in
- * continuous-agg , the partial view,
- * the materialization hypertable,
- * trigger on the raw hypertable (hypertable specified in the user view )
- * copy of the user view query (aka the direct view)
- * NOTE: The order in which the objects are dropped should be EXACTLY the same as in materialize.c"
+ * This function is intedended to be run by event trigger during CASCADE,
+ * which implies that most of the dependent objects potentially could be
+ * dropped including associated schema.
+ *
+ * These objects are: the user view itself, the catalog entry in continuous-agg,
+ * the partial view, the materialization hypertable, trigger on the raw hypertable
+ * (hypertable specified in the user view) copy of the user view query
+ * (aka the direct view).
+ *
+ * NOTE: The order in which the objects are dropped should be EXACTLY the
+ * same as in materialize.c
```

Comment:
Am I correct that the comment is just explicitly describes this assumption, which existed before this PR? Or is it brought by this PR?

Reply:
Yes, that was before. Just moved a couple of lines below close to the actual action